src/modules/mc_pos_control/PositionControl.hpp

```diff
@@ -182,42 +188,23 @@ class PositionControl
 	Controller::Constraints _constraints{}; /**< variable constraints */
 	bool _skip_controller{false}; /**< skips position/velocity controller. true for stabilized mode */
 
-	/**
-	 * Position Gains.
-	 * Pp: P-controller gain for position-controller
-	 * Pv: P-controller gain for velocity-controller
-	 * Iv: I-controller gain for velocity-controller
-	 * Dv: D-controller gain for velocity-controller
-	 */
-	matrix::Vector3f _Pp, _Pv, _Iv, _Dv = matrix::Vector3f{0.0f, 0.0f, 0.0f};
-
-	float MPC_THR_MAX{1.0f}; /**< maximum thrust */
-	float MPC_THR_HOVER{0.5f}; /** equilibrium point for the velocity controller */
-	float MPC_THR_MIN{0.0f}; /**< minimum throttle for any position controlled mode */
-	float MPC_MANTHR_MIN{0.0f}; /**< minimum throttle for stabilized mode */
-	float MPC_XY_VEL_MAX{1.0f}; /**< maximum speed in the horizontal direction */
-	float MPC_Z_VEL_MAX_DN{1.0f}; /**< maximum speed in downwards direction */
-	float MPC_Z_VEL_MAX_UP{1.0f}; /**< maximum speed in upwards direction */
-	float MPC_TILTMAX_AIR{1.5}; /**< maximum tilt for any position/velocity controlled mode in radians */
-	float MPC_MAN_TILT_MAX{3.1}; /**< maximum tilt for manual/altitude mode in radians */
-
-	// Parameter handles
-	int _parameter_sub { -1 };
-	param_t MPC_Z_P_h { PARAM_INVALID };
-	param_t MPC_Z_VEL_P_h { PARAM_INVALID };
-	param_t MPC_Z_VEL_I_h { PARAM_INVALID };
-	param_t MPC_Z_VEL_D_h { PARAM_INVALID };
-	param_t MPC_XY_P_h { PARAM_INVALID };
-	param_t MPC_XY_VEL_P_h { PARAM_INVALID };
-	param_t MPC_XY_VEL_I_h { PARAM_INVALID };
-	param_t MPC_XY_VEL_D_h { PARAM_INVALID };
-	param_t MPC_XY_VEL_MAX_h { PARAM_INVALID };
-	param_t MPC_Z_VEL_MAX_DN_h { PARAM_INVALID };
-	param_t MPC_Z_VEL_MAX_UP_h { PARAM_INVALID };
-	param_t MPC_THR_HOVER_h { PARAM_INVALID };
-	param_t MPC_THR_MAX_h { PARAM_INVALID };
-	param_t MPC_THR_MIN_h { PARAM_INVALID };
-	param_t MPC_MANTHR_MIN_h { PARAM_INVALID };
-	param_t MPC_TILTMAX_AIR_h { PARAM_INVALID };
-	param_t MPC_MAN_TILT_MAX_h { PARAM_INVALID };
+	DEFINE_PARAMETERS(
+	        (ParamFloat<px4::params::MPC_THR_MAX>) MPC_THR_MAX,
```

Comment:
I strongly suggest you stick to the usual naming pattern for members: something like `_maximum_thrust`. Otherwise we have another inconsistency in the code-base.
However it's not part of this PR.

Reply:
I discuss that with @MaEtUgR, and we came to the conclusion that it is more confusing to redefine the parameter names. For instance, the parameter in one module can have a complete different name in another module. In addition, from the name it is also not clear who is the owner of that parameters. 

Reusing the name of the metadata removes the consistent redefinition of parameter names...

Reply:
That needs a bigger upstream discussion then, you cannot simply make that decision in an unrelated PR. And if decided someone needs to go through the codebase and update all other places.
In addition, there are also reasons against this, for example:
- allows to use more descriptive variable names, since params are restricted to 16 chars.
- naming consistency with other other class members: after all, a parameter is just a class member like any other
- naming conflicts/inconsistencies with macros/globals: typically macros or globals use all uppercase. This is also confusing for newcomers

Reply:
> That needs a bigger upstream discussion then, you cannot simply make that decision in an unrelated PR

I don't expect that this [PR\(https://github.com/PX4/Firmware/pull/9253) (= base of this PR) gets merged soon, but you are right, probably these param name changes should not be added to this PR. In the meantime I will initiate a discussion about parameter naming. 

> after all, a parameter is just a class member like any other

Not really. There should be a distinction between class members that change regularly (like setpoints) and class members that are parameters which do not change frequently. 

> naming conflicts/inconsistencies with macros/globals: typically macros or globals use all uppercase. This is also confusing for newcomers

I am sure that can be easily solved by adding a px4 convention page in the px4 devguide. I think the fact that there is no formal px4 convention page is the root cause of the confusion  because a beginner has no chance to know what is legacy cold vs expected px4 implementation.

Reply:
> I am sure that can be easily solved by adding a px4 convention page in the px4 devguide. I think the fact that there is no formal px4 convention page is the root cause of the confusion because a beginner has no chance to know what is legacy code vs expected px4 implementation.

That sounds like a good idea. Probably as a sub-page of [Contribution](https://dev.px4.io/en/contribute/#contribution). All we have at the moment is that you should use astyle for formatting.

Can you create an [issue](https://github.com/PX4/Devguide/issues) outlining the main conventions you are interested in - in bullet point form and ideally with examples. We can then drag in others and iterate.